node/index.d.ts

```diff
@@ -1891,8 +1891,27 @@ declare module "dns" {
         priority: number
     }
 
-    export function lookup(domain: string, family: number, callback: (err: Error, address: string, family: number) => void): string;
-    export function lookup(domain: string, callback: (err: Error, address: string, family: number) => void): string;
+    // Supported getaddrinfo flags
+    export const ADDRCONFIG: number;
+    export const V4MAPPED: number;
+
+    export interface LookupOptions {
+        family?: number;
+        hints?: number;
+        all?: boolean;
+    }
+
+    interface Address {
+        address: string;
+        family: number;
+    }
+
+    interface QueryReqWrap {}
```

Comment:
@Flarna :+1: 
Surely it's internal. Current `lookup()` version ensures the callback to be provided so as it would be executed in async way only. Even if somebody would use result directly (in a sync-like way) they would get the only values already known to them (except those internal and deprecated like a domain):
```js
dns.lookup('example.com',function(){ console.dir(r,{colors: true} ) })
GetAddrInfoReqWrap {
  callback: { [Function: asyncCallback] immediately: true },
  family: 0,
  hostname: 'example.com',
  oncomplete: [Function: onlookup],
  domain:
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
```